AdyenCard/Utilities/CardType/CardTypeProvider.swift

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2020 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// Provide cardType detection based on BinLookup API.
+/// Fall back to local regex-based detector if API not available or BIN too short.
+internal final class CardTypeProvider: Component {
+    
+    private static let MinBinLength = 7
+    
+    private let cardTypeDetector: CardTypeDetector
+    private let apiClient: APIClientProtocol?
+    private let privateSupportedCardTypes: [CardType]
+    private var privateBinLookupService: BinLookupService?
```

Comment:
is the `private` prefix needed? in `privateBinLookupService`

Reply:
fixed